src/upload.rs

```diff
@@ -200,6 +200,47 @@ impl<'a> Uploader<'a> {
         result
     }
 
+    /// Wrapper: Delete the given file from the server.
+    pub fn delete_file(&self, file: &'a str) -> UploadResult<'a, String> {
+        UploadResult(file, self.delete(file))
+    }
```

Comment:
Do we really need this wrapper? I think we can have a single function called `delete`. What do you think?

Reply:
I really tried, but gave up and used the wrapper.

I got the following messages, because of the return type (but the calling function expects an UploadResult, so I can't just change it):

```
function should return `Result` or `Option` to accept `?`

cannot use the `?` operator in a method that returns `upload::UploadResult<'_, std::string::String>`

help: the trait `std::ops::FromResidual<std::result::Result<std::convert::Infallible, error::Error>>` is not implemented for `upload::UploadResult<'_, std::string::String>`
```

Reply:
Ah, got it. It makes sense to use a wrapper then.

Reply:
Well, I tried to do a bit of research but the FromResidual trait is experimental. However, I still would like to know how would someone solve an issue like this.

Some of the Rust idiology is still foreign to me. I try my best to learn and figure things out on my own, but sometimes it'd be nice to have a teacher... ;-)

Reply:
Frankly I'd use a wrapper or a let binding here as well. Or maybe a closure to avoid having 2 functions. Hmm...

Reply:
Man, I'd love to grab a beer with you... Too bad you and I are half around the world.

I grew up in Vienna, and there are a lot of Turkish people. My un is the head of the univrtsiry;s institure for Arabic and Turkish. Unfortunately, I never learned either of them. I was busy with Latin, French, Spanish, and of course German and English.

Reply:
> Man, I'd love to grab a beer with you... Too bad you and I are half around the world.

Yeah, that would be really nice! Just let me know if you're coming to [one of these events](https://gist.github.com/orhun/fb70e423f83b90fdad4f0caec4edca33) that I'm planning to go.

> I grew up in Vienna, and there are a lot of Turkish people

Wow, that's a lot of languages :D I actually visited Vienna once and loved it 💖 Hopefully I can visit again in the future.

Reply:
I will be in Europe for some time, but I can't say whether I'll make it to any of the events you are attending.